### ISSUE TITLE
Issues with advanced player restoration from picture in picture

### ISSUE THREAD

**Issue by defagos** [SRGSSR/srgmediaplayer-apple#77]
When returning from picture in picture, restoration occurs erratically. Sometimes it works, sometimes it interrupts playback, sometimes the animation is ugly.

### Issue type

Incorrect behavior

### Environment information

* Library version: 3.0.0, but affects prior versions as well.
* iOS version: Any
* Device: Preferrably physical

### Reproducibility

Always reproducible

### Steps to reproduce

1. Open the demo application on an iPad with PiP support.
1. Open the advanced player demo.
1. Switch to PiP.
1. Tap on the restore button and observe the transition. Repeat several times (resume playback in between if it was paused).

**Comment by defagos** [comment]
I had a rough time understanding why this was not working, while it was working in other projects based on SRG Media Player perfectly fine (e.g. Letterbox or Play). The fact that `AVPlayerViewController`-based examples were working fine in the SRG Media Player demo was also proof it was not an OS or device issue.

The implementation of PiP in the demo is in fact perfectly fine. What is not is the that the modal presentation is made using standard iOS presentation animations, internally implemented using `UIPercentDrivenInteractiveTransition`. Such transitions mess with `timeOffset`, thus altering the time of the player, even pausing it sometimes.

`AVPlayerViewController` works fine because it uses its own private transitioning delegate (`AVTransitionController`).

I therefore updated the advanced player example to use the modal transition we had on the segment player example. I removed it from this less interesting example.

I also updated `SRGMediaPlayerController` header picture in picture documentation to mention this fact.
